Bump pgpool (v4.2.4), pgbouncer (v.1.16) / md5 passwords (#2499)
epiphany-platform/epiphany: docs/home/howto/DATABASES.md

## PostgreSQL passwords encryption

> Epiphany sets up MD5 password encryption by default. Although PostgreSQL since version 10 is able to use SCRAM-SHA-256 password encryption, Epiphany does support this encryption since recommended production configuration uses more than one database host with HA configuration (repmgr) cooreating with PgBouncer and Pgpool. Pgpool is not able to parse SCRAM-SHA-256 hashes list while this encryption is enabled. Due to limited Pgpool authentication options, it is not possible to maintain the [pool_passwd file](https://www.pgpool.net/docs/42/en/html/auth-methods.html#AUTH-SCRAM) securely, hence it is also not possible to establish a secure connection between components.

Reviewer: ```suggestion
Epiphany sets up MD5 password encryption. Although PostgreSQL since version 10 is able to use SCRAM-SHA-256 password encryption, Epiphany does not support this encryption since recommended production configuration uses more than one database host with HA configuration (repmgr) cooperating with PgBouncer and Pgpool. Pgpool is not able to parse SCRAM-SHA-256 hashes list while this encryption is enabled. Due to limited Pgpool authentication options, it is not possible to maintain the [pool_passwd file](https://www.pgpool.net/docs/42/en/html/auth-methods.html#AUTH-SCRAM) securely, hence it is also not possible to establish a secure connection between components.
```

> Epiphany sets up MD5 password encryption. Although PostgreSQL since version 10 is able to use SCRAM-SHA-256 password encryption, Epiphany does not support this encryption method since recommended production configuration uses more than one database host with HA configuration (repmgr) cooperating with PgBouncer and Pgpool. Pgpool is not able to parse SCRAM-SHA-256 hashes list while this encryption is enabled. Due to limited Pgpool authentication options, it is not possible to refresh the [pool_passwd file](https://www.pgpool.net/docs/42/en/html/auth-methods.html#AUTH-SCRAM) file automatically, hence it is also not possible to establish a secure connection between components.

Reviewer: ```suggestion
Epiphany sets up MD5 password encryption. Although PostgreSQL since version 10 is able to use SCRAM-SHA-256 password encryption, Epiphany does not support this encryption method since recommended production configuration uses more than one database host with HA configuration (repmgr) cooperating with PgBouncer and Pgpool. Pgpool is not able to parse SCRAM-SHA-256 hashes list while this encryption is enabled. Due to limited Pgpool authentication options, it is not possible to refresh the [pool_passwd](https://www.pgpool.net/docs/42/en/html/auth-methods.html#AUTH-SCRAM) file automatically.
```

Author: Changed

> Because of these components MD5 password encryption is set up and is not configurable in Epiphany.

Reviewer: ```suggestion
For this reason, MD5 password encryption is set up and this is not configurable in Epiphany.
```

## PostgreSQL passwords encryption

> Epiphany sets up MD5 password encryption by default. Although PostgreSQL since version 10 is able to use SCRAM SHA-256 password encryption, Epiphany does support this encryption only for Postgresql configuration without HA. HA components like pgbouncer and pgpool are not able to refresh users/passwords list while this encryption is enabled. So those components can be used only with MD5 passwords.

Reviewer 2: ```suggestion
Epiphany sets up MD5 password encryption by default. Although PostgreSQL since version 10 is able to use SCRAM-SHA-256 password encryption, Epiphany does support this encryption only for Postgresql configuration without Pgpool. Pgpool is not able to parse SCRAM-SHA-256 hashes list while this encryption is enabled. Due to limited Pgpool authentication options, it is not possible to maintain the [pool_passwd file](https://www.pgpool.net/docs/42/en/html/auth-methods.html#AUTH-SCRAM) securely, hence it is also not possible to establish a secure connection between components.

Postgresql + Pgbouncer with SCRAM-SHA-256 deployment is possible, but it will require additional manual steps to perform in Pgboucer configuration.
```
